pkg/vpcmodel/connectivityAbstraction.go

```diff
@@ -152,8 +120,66 @@ func mergeConnectivityWithNodeSetAbstraction(
 		for _, conns := range nodeConns {
 			allConns = allConns.Union(conns)
 		}
-		res.updateAllowedConnsMap(src, nodeSet, allConns)
+		mergedConnectivity.updateAllowedConnsMap(src, nodeSet, allConns)
 	}
+	return mergedConnectivity
+}
 
-	return res
+// nodeSetAbstractionInformation() collects abstraction information of the nodeSet.
+// for now, it collects the connections that are missing for full abstraction.
+func (nsa *NodeSetAbstraction) nodeSetAbstractionInformation(mergedConnectivity GeneralConnectivityMap,
+	nodeSetToNodeSet, otherFromNodeSet, otherToNodeSet GeneralConnectivityMap,
+	nodeSet NodeSet) *AbstractionInfo {
+	abstractionInfo := &AbstractionInfo{}
+	abstractionInfo.missingEgressConnections = nsa.missingConnections(otherFromNodeSet, mergedConnectivity, nodeSet, false)
+	abstractionInfo.missingIngressConnections = nsa.missingConnections(otherToNodeSet, mergedConnectivity, nodeSet, true)
+	abstractionInfo.missingIngressConnections.addMap(nsa.missingConnections(nodeSetToNodeSet, mergedConnectivity, nodeSet, true))
+	return abstractionInfo
+}
+
+// missingConnections() is called on each of the last three groups.
+// it looks for the connections that are not exist in the group, but reflated in the mergedConnMap
+func (nsa *NodeSetAbstraction) missingConnections(connMap, mergedConnMap GeneralConnectivityMap,
+	nodeSet NodeSet, isIngress bool) GeneralConnectivityMap {
+	missingConnection := GeneralConnectivityMap{}
+	for node1, conns := range connMap {
+		// here we iterate over the nodes in the nodeSet, and not over the conns, because we can not know if conns holds the nodes:
+		for _, node2 := range nodeSet.Nodes() {
+			var nodeConnection, mergedConnection *connection.Set
+			if nodeConnection = conns[node2]; nodeConnection == nil {
+				nodeConnection = NoConns()
+			}
+			if isIngress {
+				mergedConnection = mergedConnMap[node1][nodeSet]
+			} else {
+				mergedConnection = mergedConnMap[nodeSet][node1]
+			}
+			if !nodeConnection.Equal(mergedConnection) {
+				missingConn := mergedConnection.Subtract(nodeConnection)
+				missingConnection.updateAllowedConnsMap(node1, node2, missingConn)
+			}
+		}
+	}
+	return missingConnection
+}
+
+// AbstractionInfo is the abstraction information of one nodeSet
```

Comment:
yes, LB is a nodeset, however:
NodeSetAbstraction is more general - it can work not only on load balancer, but any nodeset that we have.
I didnt made it LB specific so it might be useful later.
this file does not have the words Load balancer in it